src/main/java/com/microsoft/aad/msal4j/AcquireTokenSilentSupplier.java

```diff
@@ -39,7 +41,12 @@ AuthenticationResult execute() throws Exception {
                 clientApplication.getServiceBundle().getServerSideTelemetry().incrementSilentSuccessfulCount();
             }
 
-            if (silentRequest.parameters().forceRefresh() || StringHelper.isBlank(res.accessToken())) {
+            //Determine if the current token needs to be refreshed according to the refresh_in value
+            long currTimeStampSec = new Date().getTime() / 1000;
+            boolean afterRefreshOn = res.refreshOn() != null && res.refreshOn() > 0 &&
```

Comment:
If `res` was null, then a null pointer would happen a few lines before when it checks `res.accessToken`(), and that check has been there since May. The method that makes the auth result (`TokenCache.getCachedAuthenticationResult()`) returns an auth result made from the builder, so even if nothing in the result is set it still doesn't return a null. Just to be safe though (in case some exception somewhere could lead to a null), I've copied the 'cache miss' null check/error message that's used before returning `res` to before `res` is first referenced.